p2p/net/conn/listen.go

```diff
@@ -144,9 +145,9 @@ func Listen(ctx context.Context, addr ma.Multiaddr, local peer.ID, sk ic.PrivKey
 		Listener: ml,
 		local:    local,
 		privk:    sk,
-		cg:       ctxgroup.WithContext(ctx),
+		proc:     goprocessctx.WithContext(ctx),
 	}
-	l.cg.SetTeardown(l.teardown)
+	l.proc.SetTeardown(l.teardown)
```

Comment:
This can't be fixed. `l.teardown` is available for access only after `l` has been initialized.